jgiven-core/src/test/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportModelVisitorTest.java

```diff
@@ -0,0 +1,163 @@
+
+        ScenarioCaseModel scenarioCaseOne = new ScenarioCaseModel();
+        scenarioCaseOne.addStep(stepModel);
+
+        ScenarioModel currentScenarioModel = new ScenarioModel();
+        currentScenarioModel.addCase(scenarioCaseOne);
+        currentScenarioModel.addCase(new ScenarioCaseModel());
+
+        ReportModel reportModel = new ReportModel();
+        reportModel.addScenarioModel(currentScenarioModel);
+
+        // when
+        reportModel.accept(reportModelVisitor);
+
+        // then
+        assertThat(reportModelVisitor.getResult())
+                .isEqualTo(ImmutableList.of(
+                        "convertFeatureHeaderBlock",
+                        "convertScenarioHeaderBlock",
+                        "convertCaseHeaderBlock",
+                        "convertFirstStepBlock",
+                        "convertCaseHeaderBlock",
+                        "convertScenarioFooterBlock"));
+    }
+
+    @Test
+    public void visitTheReportModelWithTwoSections() {
+        // given
+        final StepModel sectionOne = new StepModel();
+        sectionOne.setIsSectionTitle(true);
+        sectionOne.addWords(new Word("First Section"));
+        StepModel givenStep = new StepModel();
+        givenStep.addIntroWord(Word.introWord("Given"));
+        givenStep.addWords(new Word("some"), new Word("state"));
+
+        final StepModel sectionTwo = new StepModel();
+        sectionTwo.setIsSectionTitle(true);
+        sectionTwo.addWords(new Word("Second Section"));
+        StepModel whenStep = new StepModel();
+        whenStep.addIntroWord(Word.introWord("When"));
+        whenStep.addWords(new Word("some"), new Word("action"));
+
+        ScenarioCaseModel scenarioCaseOne = new ScenarioCaseModel();
+        scenarioCaseOne.addStep(sectionOne);
+        scenarioCaseOne.addStep(givenStep);
+        scenarioCaseOne.addStep(givenStep);
+        scenarioCaseOne.addStep(sectionTwo);
+        scenarioCaseOne.addStep(whenStep);
+
+        ScenarioModel currentScenarioModel = new ScenarioModel();
+        currentScenarioModel.addCase(scenarioCaseOne);
+        currentScenarioModel.addCase(new ScenarioCaseModel());
+
+        ReportModel reportModel = new ReportModel();
+        reportModel.addScenarioModel(currentScenarioModel);
+
+        // when
+        reportModel.accept(reportModelVisitor);
+
+        // then
+        assertThat(reportModelVisitor.getResult())
+                .isEqualTo(ImmutableList.of(
+                        "convertFeatureHeaderBlock",
+                        "convertScenarioHeaderBlock",
+                        "convertCaseHeaderBlock",
+                        "convertFirstStepBlock",
+                        "convertStepBlock",
+                        "convertFirstStepBlock",
+                        "convertCaseHeaderBlock",
+                        "convertScenarioFooterBlock"));
+    }
+
+    private static class MyFakeReportBlockConverter implements ReportBlockConverter {
```

Comment:
I am curious why you chose this implementation over just using mockito. Is there a specific reason?

Reply:
Mockito was not part of the test dependencies of the `jgiven-core` project and I didn't want to add it if it wasn't already.
I presumed there was a reason for it to not be added. So I implemented my own _order of invocation verifier_.